Add elixir analyzer comment for no-rescue (#2250)
exercism/website-copy: analyzer-comments/elixir/solution/no_rescue.md

# no rescue

> Do not use exceptions for flow control. Try to solve this exercise without using `rescue`.

Reviewer: ```suggestion
Do not use exceptions for flow control.
Try to solve this exercise without using `rescue`.
```

> If you're using a standard library function whose name ends with `!` (raises errors), find an equivalent function.

Reviewer: ```suggestion
If you are using a standard library function with a name ending with a `!` (raises errors), look for an equivalent function without the `!` (often return `{:ok, result}` or `{:error, message}`).
```

> Note: If you're seeing this automated comment on your solution, the other automated comments might not be accurate. Fix this problem first!

Reviewer: ```suggestion
Note: if you are seeing this automated comment on your solution, other automated comments might not be accurate.
Fix this problem first!
```

```elixir
# preferred: for example `case`

case Date.new(1999, 13, 1) do
  {:ok, date} -> date
  {:error, _} -> nil
end

> # discouraged: rescuing exceptions

Reviewer: Should we also mention that something like this would raise
```elixir
{:ok, date} = Date.new(1999, 13, 1)
```
Or does it feel like a different topic?

Author: It does feel like a different topic. This comment is only triggered by usages of `rescue`, it doesn't really matter how the error is raised... the goal is to prevent `rescuing`.
